terraform/modules/utils/scripts/EhrHardDeletion.py

```diff
@@ -1,11 +1,59 @@
 import json
 import boto3
 import os
+import time
+from boto3.dynamodb.conditions import Key
 
 def lambda_handler(event, context):
-    print("This is the ehr-hard-deletion-lambda!")
+    dynamodbTable, inboundConversationId = parseEvent(event)
+    deleteEhrFromS3(inboundConversationId)
+    verifyDBTableRecordsDeleted(dynamodbTable, inboundConversationId)
 
-    return {
-        'statusCode': 200,
-        'body': json.dumps('Completed!'),
-    }
+def deleteEhrFromS3(inboundConversationId):
+    s3 = boto3.resource('s3')
+    try:
+        print("Retrieving S3_REPO_BUCKET environment variable")
+        s3BucketName = os.environ["S3_REPO_BUCKET"]
+        print("S3_REPO_BUCKET=" + s3BucketName)
+        print("Retrieving S3 Bucket")
+        repoBucket = s3.Bucket(s3BucketName)
+    except KeyError as error:
+        print("Failed to get S3_REPO_BUCKET environment variable: " + error)
+    except Exception as error:
+        print("Failed to find the S3 Bucket: " + error)
+
+    if any(True for _ in repoBucket.objects.filter(Prefix=inboundConversationId + "/")): # Checks to see objects exist in the S3 Bucket for the given inboundConversationId
```

Comment:
this is a bit hard to read
would suggest 

`if list(repoBucket.objects.filter(Prefix=inboundConversationId + "/")):
    .....`

This puts all the filtered objects into a list; the if statement is true if the list is not empty and false if the list is empty.